src/tree-views/expansionstaterestorer.cpp

```diff
@@ -58,13 +58,13 @@ void ExpansionStateRestorer::addChildrenIndexes(const QModelIndex &parent)
 {
     for (int i = 0; i < m_model->rowCount(parent); ++i)
         for (int j = 0; j < m_model->columnCount(parent); ++j)
-            tryAddIndex(parent.child(i, j));
+            tryAddIndex(m_model->index(i, j, parent));
```

Comment:
wow, what's the difference?

Reply:
The difference is that in case of `m_model->index(i, j, parent)` passing default constructed `QModelIndex()` object allows to iterate through whole tree, starting from the root node. This is due to how model behaves:

```
const Node *Model::dataNode(const QModelIndex &index) const
{
    return index.isValid() ? static_cast<Node*>(index.internalPointer()) : m_root;
}
```

Removed version, to iterate over whole tree, would require passing as parent something like `QModelIndex(m_model->modelRoot)` . 

Hoped I used it correctly :)